src/Interpreters/Cache/FileSegment.cpp

```diff
@@ -99,6 +99,7 @@ size_t FileSegment::getFirstNonDownloadedOffsetUnlocked(std::unique_lock<std::mu
     return range().left + getDownloadedSizeUnlocked(segment_lock);
 }
 
+/// same as getFirstNonDownloadedOffset ?
```

Comment:
Yes, it is the same with current code, but it will not be the same after https://github.com/ClickHouse/ClickHouse/pull/40026 is merged. I extracted refactoring of cache code into another pr and merged it some time ago because of the number of conflicts arising all the time, so currently it might be a little confusing that there are two similar methods.